tests/ixia/ecn/files/helper.py

```diff
@@ -259,12 +233,15 @@ def __config_capture_ip_pkt(testbed_config, port_id):
     """
 
     """ We only capture IP packets """
-    ip_filter = CustomFilter(filter='40', mask='0f', offset=14)
-    result = [Capture(name='rx_capture',
-                      port_names=[testbed_config.ports[port_id].name],
-                      choice=[BasicFilter(ip_filter)],
-                      enable=True)]
-    return result
+    cap = testbed_config.captures.capture(name='rx_capture')[-1]
+    cap.port_names = [testbed_config.ports[port_id].name]
+    cap.format = cap.PCAP
+
+    ip_filter, = cap.filters.filter()
+    ip_filter.custom.value = '40'
+    ip_filter.custom.offset = 14
+    ip_filter.custom.mask = '0f'
+
 
 def __run_traffic(api,
```

Comment:
This function is filled with API calls yet no asserts are used for error checking. We should check for errors and raise it, and if there's a warning, log it.